activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -105,7 +105,7 @@ def build(attributes = nil, &block)
         if attributes.is_a?(Array)
           attributes.collect { |attr| build(attr, &block) }
         else
-          add_to_target(build_record(attributes, &block))
+          add_to_target(build_record(attributes, &block), replace: true)
```

Comment:
Replace uses `Array#index` which uses `==` to compare, but that seems safe in this context because active record's `==` uses `===` unless an ID is present. There's also no test coverage that disagrees with doing this, so 👍

Reply:
I haven't fully figured it out, but we had a scenario where we override `<=>` (but not `==`) on a model, and this change made calling `.build` for that model really really slow. Any chance `Array#index` calls `<=>`?